src/convertToObject.js

```diff
@@ -3,16 +3,37 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const splitExpressions = sourceString
+    .split(';')
+    .join('$')
+    .split(':')
+    .join('$')
+    .split('$');
+
+  const trimmedExpressions = splitExpressions.map(x => x.trim());
+
+  const codeExpressions = trimmedExpressions.filter(x => x.length > 0);
+
+  const entries = codeExpressions.reduce((rows, key, index) => (index % 2 === 0
+    ? rows.push([key])
+    : rows[rows.length - 1].push(key)) && rows, []);
```

Comment:
it's not very clear what it does, maybe just try to create an empty object and then assign the key and values to it like `sourceObject[key] = value;` in a loop